ksqldb-common/src/main/java/io/confluent/ksql/util/DecimalUtil.java

```diff
@@ -113,16 +113,20 @@ public static int precision(final Schema schema) {
   }
 
   /**
-   * Formats the decimal string, padding zeros if necessary.
+   * Formats the decimal string, adding trailing zeros if necessary.
    *
    * @param value the value
    * @return the formatted string
    */
   public static String format(final int precision, final int scale, final BigDecimal value) {
     final DecimalFormat format = new DecimalFormat();
-    format.setMinimumIntegerDigits(precision - scale);
     format.setMinimumFractionDigits(scale);
 
+    // Avoids adding leading zeros if precision is higher than the value precision
+    if (precision < value.precision()) {
+      format.setMinimumIntegerDigits(precision - scale);
+    }
```

Comment:
why do we need this at all? does decimal format not handle integer digits by default?

Reply:
I wasn't sure why this was needed before, but I know now this is used to add the zeros at the left. I removed it now.